Docs/04-GeneratingCertificates.md

```diff
@@ -0,0 +1,103 @@
+# Generating Certificates
+
+This document will walk you through one way to generate a CA and self-signed certificate for use with Steward.  
+
+Two-way trust must be established between Steward and the Strategy Provider client for each request. To accomplish this, both the Server and Client will need TLS certificates and the CA certificate of the other party (Steward needs the client's CA, and the client needs Steward's CA). To make generating Steward's certificates easier, we've provided a convenience script called `gen-script.sh` which can be found [here](https://github.com/peggyjv/steward/gen-script.sh). The script takes one optional flag for the output location of the certificates and their signing keys. 
+
+## Steps
+
+0. Ensure you have OpenSSL installed.
+
+This script requires OpenSSL to be installed; it will not work with LibreSSL.
+
+Note that LibreSSL still uses the binary name `openssl`. You'll know if you have the wrong binary if `openssl version` says "LibreSSL".
+
+1. Run the following command from the root of this repository
+
+```
+./gen-certs.sh -o some/output/location/
+```
+
+If you don't set the output path flag, certificates and keys will be created in the current working directory.
+
+2. Fill out the Server CA information
+
+After running this command you run through a series of prompts. The first will be for information pertaining to the certificate authority like geographical location, email, etc. These fields can be safely left blank. 
```

Comment:
They might want to fill something out for CN or OU just to have some identifying information as to what the cert is for.